domain/src/main/java/org/oppia/android/domain/state/StateDeck.kt

```diff
@@ -108,6 +114,8 @@ internal class StateDeck internal constructor(
     currentDialogInteractions.clear()
     hintList.clear()
     pendingTopState = state
+    revealedHintIndex = -1
```

Comment:
Add a comment as to why the hint index is -1 here

Reply:
done